ps1/ps1/src/twitter/Extract.java

```diff
@@ -1,177 +0,0 @@
-
-import java.util.List;
-import java.util.Set;
-import java.util.HashSet;
-import java.time.Instant;
-import twitter.Timespan;
-import java.lang.StringBuilder;
-
-
-/**
- * Extract consists of methods that extract information from a list of tweets.
- * 
- * DO NOT change the method signatures and specifications of these methods, but
- * you should implement their method bodies, and you may add new public or
- * private methods or classes if you like.
- */
-public class Extract {
-
-    /**
-     * Get the time period spanned by tweets.
-     * 
-     * @param tweets
-     *            list of tweets with distinct ids, not modified by this method.
-     * @return a minimum-length time interval that contains the timestamp of
-     *         every tweet in the list.
-     */
-    public static Timespan getTimespan(List<Tweet> tweets) {
-        
-        if (tweets.size() < 1) {
-            throw new RuntimeException("Tweet list doesn't fit spec.");
-        }
-        
-        // set the starting min/max
-        Instant min = tweets.get(0).getTimestamp();
-        Instant max = tweets.get(0).getTimestamp();
-        
-        // replace min and max time if min-er or max-er time is found
-        for (int i = 1; i < tweets.size(); i++){
-            
-            Instant current = tweets.get(i).getTimestamp();
-            if (current.isAfter(max)) {
-                max = current;
-                }
-            if (current.isBefore(min)) {
-                min = current;
-                }
-        }
-                
-        Timespan span = new twitter.Timespan(min, max);
-                        
-        return span;
-        
-    }
-    
-    /**
-     * Find and return the next username from position at_char in a String text.
-     * 
-     * @param text of a tweet (string)
-     *        at_char is the index from which to search the next username
-     * @return the next username, converted to lowercase String. If none, returns
-     * an empty String.
-     */
-    public static String getUsername(String text, int at_char) {
-        
-        String AcceptableChars = 
-                "abcdefghijklmnopqrstuvwxyz"
-              + "ABCDEFGHIJKLMNOPQRSTUVWXYZ"
-              + "0123456789"
-              + "-_" ;
-        
-        char char_preceding;
-        char char_succeeding;
-
-        if(at_char == -1){
-            throw new RuntimeException("@ symbol doesn't appear in tweet.");
-            
-        } else if(at_char >= text.length()) {
-            throw new RuntimeException("character index out of range of text length.");
-            
-        } else if(at_char == 0){
-            // if it's the first character of the tweet,
-            // we add an unacceptable preceding character to avoid errors
-            char_preceding = '#';
-            char_succeeding = text.charAt(at_char+1);
-            
-        } else if (at_char == text.length()-1) {
-            // if it's the last character of the tweet,
```

Comment:
if `at_char` is the last character, shouldn't this just end ?

Reply:
....yes haha